lib/_http_outgoing.js

```diff
@@ -125,6 +127,19 @@ function OutgoingMessage() {
 }
 util.inherits(OutgoingMessage, Stream);
 
+Object.defineProperty(OutgoingMessage.prototype, 'socket', {
+  get: function() {
+    return this[kSocket];
+  },
+  set: function(socket) {
+    this[kSocket] = socket;
+    if (socket && socket[async_id_symbol] === undefined) {
+      socket[async_id_symbol] = async_hooks.newUid();
```

Comment:
This is an issue I've been wresting with for the last month while I was working on https://github.com/nodejs/node/pull/14208. The problem with assigning an `async_id` to the `OutgoingMessage` is because `OutgoingMessage` isn't actually asynchronous in any way, so it will never emit a `before()`/`after()`/`destroy()`. So we're only assigning an `async_id` as a stop gap.

There are two long term solutions that I could implement quickly:

1. Assign the `this[async_id_symbol] = newUid()` in the `OutgoingMessage` constructor. Then use https://github.com/nodejs/node/pull/14208 to propagate that value to the socket when it's assigned. This will require detecting if the `socket` will call the `destroy()` callback, or if it needs to be queued to be called in the `'finish'` event. For this to happen I'll need sign-off on said PR.

2. I've also considered and experimented with assigning error codes to the `async_id_symbol`/`trigger_id_symbol` that can be safely handled. In this case I'd assign `UV_ENODATA` in the constructor since the `socket` hasn't been created yet, then the `async_id_symbol` can be assigned when the `socket` has been created. This has the drawback of breaking async call graphs, but is easier to implement and doesn't require the additional PR.

I'm going to step out and clear my head. Give me feedback and I can get a fix posted.